app/screens/channel_add_people/channel_add_people.tsx

```diff
@@ -0,0 +1,340 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useReducer, useRef, useState} from 'react';
+import {defineMessages, useIntl} from 'react-intl';
+import {Keyboard, Platform, StyleSheet, View} from 'react-native';
+import {SafeAreaView} from 'react-native-safe-area-context';
+
+import {addMembersToChannel} from '@actions/remote/channel';
+import {fetchProfilesNotInChannel, searchProfiles} from '@actions/remote/user';
+import Loading from '@components/loading';
+import Search from '@components/search';
+import SelectedUsers from '@components/selected_users_panel';
+import UserList from '@components/user_list';
+import {General} from '@constants';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {ChannelModel} from '@database/models/server';
+import {debounce} from '@helpers/api/general';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import {t} from '@i18n';
+import {popTopScreen, setButtons} from '@screens/navigation';
+import {alertErrorWithFallback} from '@utils/draft';
+import {changeOpacity, getKeyboardAppearanceFromTheme} from '@utils/theme';
+import {filterProfilesMatchingTerm} from '@utils/user';
+
+const ADD_BUTTON = 'add-button';
+
+const close = () => {
+    Keyboard.dismiss();
+    popTopScreen();
+};
+
+const style = StyleSheet.create({
+    container: {
+        flex: 1,
+    },
+    searchBar: {
+        marginLeft: 12,
+        marginRight: Platform.select({ios: 4, default: 12}),
+        marginVertical: 12,
+    },
+});
+
+const messages = defineMessages({
+    error: {
+        id: t('mobile.channel_add_people.error'),
+        defaultMessage: 'We could not add those users to the channel. Please check your connection and try again.',
+    },
+    button: {
+        id: t('mobile.channel_add_people.title'),
+        defaultMessage: 'Add Members',
+    },
+});
+
+function reduceProfiles(state: UserProfile[], action: {type: 'add'; values?: UserProfile[]}) {
+    if (action.type === 'add' && action.values?.length) {
```

Comment:
If we only have one action (`add`) we may not need the reducer. We can use only the `setState` function passing a function. i.e.:

```
const [profiles, setProfiles] = useState([]);
...
setProfiles((prev) => [...prev, ...values]);
```